adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1718,14 +1796,34 @@ private boolean stopCalled() {
         }
 
         /**
-         * Closes a connection to a client.
+         * Actively closes the connection to the device.
+         *
+         * @param reason The reason for closing the connection.
+         * @throws NullPointerException if reason is {@code null}.
+         */
+        protected final void close(final String reason) {
+            Objects.requireNonNull(reason);
+            endpoint.closeHandler(v -> {});
+            final Span span = newSpan("close device connection");
+            span.log("close device connection due to " + reason);
+            onCloseInternal(span, false);
```

Comment:
This means that we do not send the TTD event nor the disconnected event, doesn't it? If so, why don't we send the events in this case?

Reply:
Sending an event in case the corresponding tenant/device is disabled/deleted would fail (*). Therefore sending the events is skipped here.
I've just committed a change where this is made clearer in the code.

(*) It wouldn't have to fail in case the `LoggingConnectionEventProducer` is used (although in that case there is still a "getTenant" invocation being done which would fail if the tenant got deleted). I think a change to enable invoking that event producer in the case here could be done in a subsequent PR, along with necessary changes to the `ConnectionEventProducer` interface.